packages/shared/src/domain/address/zip.ts

```diff
@@ -16,8 +23,16 @@ export function normalizeZipCode(zipCode: string): string {
   return trimmedZip.slice(0, 5);
 }
 
-function isValidZipCode(zipCode: string): boolean {
-  if (zipCode.length === 0) return false;
-  const regex = /^[0-9-]+$/;
-  return regex.test(zipCode);
+export function normalizeZipCodePassive(zipCode: string): string | undefined {
+  const trimmedZip = zipCode.trim();
+  if (!isValidZipCode(trimmedZip)) return undefined;
+  if (trimmedZip.includes("-") && trimmedZip.trim().length === 9) return trimmedZip.slice(0, 4);
+  if (trimmedZip.trim().length === 8) return trimmedZip.slice(0, 4);
+  return trimmedZip.slice(0, 5);
+}
```

Comment:
Instead of duplicating the function, we could make `normalizeZipCode` use this one and throw on undefined, like you done for `normalizeZipCodeStrict`

Reply:
WRT to naming, we've been using `Safe` as a prefix for alternative functions that don't throw.

Reply:
Done.